consumer_group.go

```diff
@@ -126,14 +126,17 @@ func newConsumerGroup(groupID string, client Client) (ConsumerGroup, error) {
 		return nil, ConfigurationError("consumer groups require Version to be >= V0_10_2_0")
 	}
 
-	consumer, err := NewConsumerFromClient(client)
-	if err != nil {
-		return nil, err
+	c := &consumer{
+		client:          client,
+		conf:            client.Config(),
+		children:        make(map[string]map[int32]*partitionConsumer),
+		brokerConsumers: make(map[*Broker]*brokerConsumer),
+		metricRegistry:  newCleanupRegistry(client.Config().MetricRegistry),
```

Comment:
Thanks for catching this problem. Was there a specific reason why you in-lined the consumer struct initialisation rather than calling `newConsumer` as-is? We're just in one big `sarama` package so it doesn't matter that its not exported

```suggestion
	c, err := newConsumer(client)
	if err != nil {
		return nil, err
	}
```